api/src/test/java/org/openmrs/api/OrderSetServiceTest.java

```diff
@@ -337,6 +336,36 @@ public void shouldRetireOrderSetAndOrderSetMembersAsWell() throws Exception {
 			assertTrue(orderSetMember.getRetired());
 		}
 	}
+	
+	@Test
+	public void retireOrderSetWithNullRetireReason_shouldThrowException() throws Exception {
+		expectedException.expect(IllegalArgumentException.class);
+		expectedException.expectMessage("retire reason cannot be empty or null");
+		executeDataSet(ORDER_SET);
+		OrderSet orderSet = orderSetService.getOrderSet(2001);
+		orderSetService.retireOrderSet(orderSet, null);
+		Context.flushSession();
+	}
+	
+	@Test
+	public void retireOrderSetWithWhiteSpaceRetireReason_shouldThrowException() throws Exception {
+		expectedException.expect(IllegalArgumentException.class);
+		expectedException.expectMessage("retire reason cannot be empty or null");
+		executeDataSet(ORDER_SET);
+		OrderSet orderSet = orderSetService.getOrderSet(2001);
+		orderSetService.retireOrderSet(orderSet, " ");
+		Context.flushSession();
+	}
+	
+	@Test
+	public void retireOrderSetWithEmptyRetireReason_shouldThrowException() throws Exception {
+		expectedException.expect(IllegalArgumentException.class);
+		expectedException.expectMessage("retire reason cannot be empty or null");
+		executeDataSet(ORDER_SET);
+		OrderSet orderSet = orderSetService.getOrderSet(2001);
+		orderSetService.retireOrderSet(orderSet, " ");
+		Context.flushSession();
+	}
```

Comment:
May be i will leave one test (retireOrderSetWithNullRetireReason_shouldThrowException) instead of 2. Other wise we will not be unit testing the exception scenario